dashboard/src/components/dataBrowser/RuleGroupEditor/useRuleGroupEditor.ts

```diff
@@ -0,0 +1,19 @@
+import { createContext, useContext } from 'react';
+
+export const RuleGroupEditorContext = createContext<{
+  schema: string;
+  table: string;
+}>({
+  schema: '',
+  table: '',
+});
+
+export default function useRuleGroupEditor() {
+  const context = useContext(RuleGroupEditorContext);
+
+  if (!context) {
+    throw new Error('useRuleGroupEditor must be used within a RuleGroupEditor');
```

Comment:
Any reason to not name `RuleGroupEditorProvider` instead of `RuleGroupEditor` ?

Reply:
I don't think it makes sense because we are only allowing this hook to be accessed inside the boundaries of a `RuleGroupEditor`. We don't have a `RuleGroupEditorProvider`.